assets/src/less/blocks/wp-block-cover/layout.less

```diff
@@ -42,16 +39,127 @@
 			inline-size: 100%;
 			block-size: 100%;
 			opacity: 0.25;
-			z-index: 1;
+			z-index: 2;
+		}
+
+		&[style*=background-color]:before {
+			background-color:inherit;
 		}
 	}
 
 	.wp-block-cover__inner-container {
-		position: relative;
-		z-index: 2;
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		position: absolute;
+		inline-size: 100%;
+		block-size: 100%;
+		z-index: 3;
 		.stack--tight;
+
+		padding: @space-3;
+		> .has-background {
+			padding: @space-2;
+		}
 	}
 
-}
+	&.alignleft {
```

Comment:
If we're going to do this, we should do `.alignright` also and use inline-size rather than width.

Reply:
updated